source/stylesheets/_screen.scss

```diff
@@ -669,6 +674,7 @@ html, body {
   .toc-link {
     padding-top: 0.3em;
     padding-bottom: 0.3em;
+    text-decoration: none;
```

Comment:
Same as above - might not be necessary anymore with your `h2[:content]` to `h2[:title]` fix. From what I can see, there's already `text-decoration: none;` being applied to the `.toc-link` and we don't need to add this line specific to `@media (max-width: $tablet-width)`
<img width="907" alt="image" src="https://github.com/user-attachments/assets/9ed0b810-0900-4484-a138-8a3df1e8d07d">